### PR TITLE
Room custom data

### PR DESCRIPTION
**Pull request by mdpye**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,19 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
...
+
+- URL Encoding of room IDs when sending typing indicators
+- Deserialization of Room privacy (always returned false)
+- Deserialization of User customData - was Map<String, String> when other clients were not
+  restricted to string values by the API.
+- Room name is optional when updating rooms
+
+### Added
+
+- Room objects have a custom data field which can hold a map of arbitrary data,
```
Comment (vivangkumar) [inline CHANGELOG.md]: Worth adding that updates to custom data are propagated in real time?

File: chatkit-core/src/integrationTest/kotlin/com/pusher/chatkit/RoomSpek.kt [hunk 2]
```diff
@@ -138,14 +139,226 @@ class RoomSpek : Spek({
             assertThat(room.name).isEqualTo(GENERAL)
         }
 
+        it("creates private room") {
```
Comment (vivangkumar) [inline chatkit-core/src/integrationTest/kotlin/com/pusher/chatkit/RoomSpek.kt]: Nice, love the test cases!

File: CHANGELOG.md (hunk 1 above)
Comment (mdpye) [reply CHANGELOG.md]: I thought it was implied (all room updates are propagated in real time).

Let's mention it in docs and feature announcement.

**Comment by mdpye** [comment]
Re: updating individual fields, we could support that client side by merging existing data, but it's pretty easy to do

```
cUser.updateRoom(
  room = myRoom,
  customData = myRoom.customData.apply { this += "field" to "value" }
}
```

And that makes it clearer that the operation is a complete replacement on the server side.

**Comment by mdpye** [comment]
Actually, it probably needs to be `myRoom.customData.toMutableMap().apply { this += "field" to "value" }` which is getting a bit gross, but either way, we don't update rooms  by "saving" a `Room` object, so it would be a bigger change to the way the library presents update operations.

Maybe later.

File: CHANGELOG.md (hunk 1 above)
Comment (mdpye) [reply CHANGELOG.md]: Checked the docs, it already specifies that members will receive an event with the new room details.

**Pull request merged**